lib/patron.rb

```diff
@@ -0,0 +1,19 @@
+require 'yaml'
+
+class Patron
+
+  attr_accessor :bookrack, :library
+
+  def initialize
+    @patron = YAML.load_file('./lib/patrondb.yml')
+  end
+
+  def available_books
+    @patron.select { |obj| obj[:available] == true  }.each {|val| puts val }
+  end
```

Comment:
Do not use `puts` - that is for terminal commands only. I think this might work but the only reason is because that is the last thing this method sees and it returns it by default. Take out the `puts` in these methods and see if everything still runs. You should only use `puts` for things like logging and debugging.